src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -52,17 +60,32 @@ public DrsService(SnapshotService snapshotService,
                       DrsIdService drsIdService,
                       IamService samService,
                       ResourceService resourceService,
+                      ConfigurationService configurationService,
+                      KubeService kubeService,
                       PerformanceLogger performanceLogger) {
         this.snapshotService = snapshotService;
         this.fileService = fileService;
         this.drsIdService = drsIdService;
         this.samService = samService;
         this.resourceService = resourceService;
+        this.configurationService = configurationService;
+        this.kubeService = kubeService;
         this.performanceLogger = performanceLogger;
     }
 
+
+
     public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String drsObjectId, Boolean expand) {
 
+        // make sure not too many requests are being made at once
+        currentDRSRequests++;
+        int podCount = kubeService.getActivePodCount();
+        int maxDRSLookups = configurationService.getParameterValue(ConfigEnum.DRS_LOOKUP_MAX);
+        int max = maxDRSLookups / podCount;
+        if (currentDRSRequests >= max) {
+            // what's the best way to return this in our current format?
+            HttpStatus.TOO_MANY_REQUESTS;
```

Comment:
I believe that it's a matter of throwing an exception that maps to a `HttpStatus.TOO_MANY_REQUESTS`. It looks like we don't have a thing in the `GlobalExceptionHandler` to handle those types of errors yet but you could add a new Abstract `TooManyRequestsException` class and then add a handler for that in `GlobalExceptionHandler`.  Does that make sense?  For example, check out line 63, for conflict exceptions.  You could have something super similar like:
```
@ExceptionHandler(TooManyRequestsException.class)
@ResponseStatus(HttpStatus. TOO_MANY_REQUESTS)
public ErrorModel tooManyRequestsHandler(DataRepoException ex) {
    return buildErrorModel(ex, ex.getErrorDetails());
}
```

Then in this class, throw an exception where the exception class extends `TooManyRequestsException`